src/puget_cli/main.clj

```diff
@@ -24,8 +25,15 @@
                    first
                    edn/read-string)}))
 
+(defn in-seq
+  [reader]
+  (lazy-seq
+    (if-let [obj (edn/read {:eof nil} reader)]
```

Comment:
nil is a valid EDN value, probably better to take a sentinel object

Reply:
Also adding a close after returning a lazy sequence feels weird. Just make this a strict function and use `with-open`

Reply:
Good point about `nil`, I'll switch to a sentinel.

I switched around the implementation to use `with-open`. I moved the `puget/pprint` into the `with-open` so that only one EDN form is in memory at a time. Thoughts?